quickcheck/src/test/java/fj/test/TestBool.java

```diff
@@ -0,0 +1,30 @@
+package fj.test;
+
+import fj.data.test.PropertyAssert;
+import org.junit.Test;
+
+import static fj.test.Arbitrary.arbBoolean;
+import static fj.test.Arbitrary.arbitrary;
+import static fj.test.Bool.bool;
+import static fj.test.Property.property;
+
+public class TestBool {
+
+    public static final class BoolTestClass {
+        final Boolean b;
+        public BoolTestClass(Boolean b) {
+            this.b = b;
+        }
+        public Boolean getBoolean(){
+            return b;
+        }
+    }
+
+    @Test
+    public void testBool() {
+        final Arbitrary<Boolean> arbBooleanR = arbitrary(arbBoolean.gen);
+        final Property p = property(arbBooleanR, arbBooleanR, (m1, m2) -> bool(m1.equals(m2))
+                .implies(new BoolTestClass(m1).getBoolean() == new BoolTestClass(m2).getBoolean()));
```

Comment:
What is the purpose of wrapping the Boolean in a class and getting the boolean back out, rather than a direct `m1 == m2` expression?